mobile_wallet/src/lib.rs

```diff
@@ -88,6 +97,52 @@ fn make_signatures(
     (tx.signature, body)
 }
 
+/// Compute the message digest for some message and an account. The message
+/// digest is constructed so that it cannot be the prefix of an actual
+/// account transaction.
+fn get_message_digest(account_address: [u8; 32], message: String) -> [u8; 32] {
+    let prepend_bytes = [0_u8; 8];
+    let message_as_bytes = message.as_bytes();
+
+    let mut hasher = Sha256::new();
+    hasher.update(account_address);
+    hasher.update(prepend_bytes);
+    hasher.update(message_as_bytes);
+    let hash: [u8; 32] = hasher.finalize().into();
+    hash
+}
+
+fn sign_message_with_keys(
+    keys: &AccountKeys,
+    msg: String,
+    account_address: [u8; 32],
+) -> BTreeMap<CredentialIndex, BTreeMap<KeyIndex, Signature>> {
+    let message_digest = get_message_digest(account_address, msg);
+
+    // TODO: The code below has been copied from sign_transaction_hash(), where to
+    // put a shared function to avoid duplication?
+    let iter = keys
```

Comment:
This is copy-pasted, but I was unsure where to put the code to share it. Any pointers?

Reply:
Can you not use that function directly?

changing
```
fn get_message_digest(account_address: [u8; 32], message: String) -> hashes::TransactionSignHash {
```
and then 
```
let hash_to_sign = get_message_digest(...);
keys.sign(&hash_to_sign)
```
Maybe I am missing something that is blocking this?

Reply:
You are right. It does become:
```
keys.sign_transaction_hash(&hash_to_sign)
```
which is a little awkward as it is not a transaction hash, but I think it is fine.